reference/test/preconditioner/batch_jacobi_kernels.cpp

```diff
@@ -0,0 +1,189 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <ginkgo/core/preconditioner/batch_jacobi.hpp>
+
+
+#include <limits>
+
+
+#include <gtest/gtest.h>
+
+
+#include <ginkgo/core/base/batch_multi_vector.hpp>
+#include <ginkgo/core/base/exception.hpp>
+#include <ginkgo/core/base/executor.hpp>
+#include <ginkgo/core/matrix/batch_csr.hpp>
+#include <ginkgo/core/preconditioner/jacobi.hpp>
+
+
+#include "core/preconditioner/batch_jacobi_kernels.hpp"
+#include "core/test/utils.hpp"
+#include "core/test/utils/batch_helpers.hpp"
+
+
+namespace detail {
+
+
+template <typename ValueType>
+void is_equivalent_to_unbatched(
+    const gko::size_type batch_idx,
+    const std::unique_ptr<gko::batch::preconditioner::Jacobi<ValueType>>&
+        batch_prec,
+    const std::unique_ptr<gko::preconditioner::Jacobi<ValueType>>& unbatch_prec)
+{
+    const auto num_blocks = batch_prec->get_num_blocks();
+    const auto block_ptrs = batch_prec->get_const_block_pointers();
+    const auto blocks_batch_arr = batch_prec->get_const_blocks();
+    const auto blocks_unbatch_arr = unbatch_prec->get_blocks();
+    auto storage_scheme_unbatch = unbatch_prec->get_storage_scheme();
+
+    const auto tol = r<ValueType>::value;
+
+    for (int k = 0; k < num_blocks; k++) {
+        const auto bsize = block_ptrs[k + 1] - block_ptrs[k];
+        for (int r = 0; r < bsize; r++) {
```

Comment:
also here, maybe use dense matrix views.

Reply:
Is there a reason it's not possible to use the views here?

Reply:
We compare the unbatched version with the batched one, the storage of the blocks is different. Additionally, for debugging purposes, it is easier if there is a element based comparison in case there are issues.